### PR TITLE
React: Fix consistency of spacing/indentation/semicolons

### PR DESCRIPTION
**Pull request by MaoShizhong**
## Because
Noticed inconsistency across some of the React lessons with spacing/indentation/semicolon usage. Many of these have been fixed so semicolons are used where they can be put, and indentation is kept to 2 spaces as with the other lessons.

## This PR
- Fixes indentation consistency to 2-spaces
- Fixes missing space after function params parens
- Adds in semicolons to keep usage consistent across lessons
- Trailing white space automatically removed

## Issue
N/A

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [ ] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: react/states_and_effects/more_on_state.md [hunk 1]
```diff
@@ -169,34 +169,34 @@ There are native HTML elements that maintain their own internal state. The `inpu
...
+1. Read the following articles from the React documentation:
+  - [State as a Snapshot](https://react.dev/learn/state-as-a-snapshot)
+  - [Choosing the State Structure](https://react.dev/learn/choosing-the-state-structure)
+  - [Sharing State Between Components](https://react.dev/learn/sharing-state-between-components)
 
 2. Update the `Person` component we've been using above.
-   - Add two separate input fields for first name and last name. The updated full name should be displayed on every keystroke on either of the two input fields. 
-   - There are many ways you can do this. Keep in mind what you've learned in this lesson while coding it out.
+  - Add two separate input fields for first name and last name. The updated full name should be displayed on every keystroke on either of the two input fields.
+  - There are many ways you can do this. Keep in mind what you've learned in this lesson while coding it out.
```
Comment (rlmoser99) [inline react/states_and_effects/more_on_state.md:199]: These will need to be reverted back to 3 spaces, according to our style guide.
> a 3 space indent when nesting inside a numbered list
https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md#nested-lists

**Review by rlmoser99 — CHANGES_REQUESTED** [review]
Thanks for taking the time to make these consistency improvements! I greatly appreciate how much more consistent these lessons will be! Just got one small change, that is specifically called out in our style guide.

File: react/states_and_effects/more_on_state.md (hunk 1 above)
Comment (MaoShizhong) [reply react/states_and_effects/more_on_state.md]: That's my bad! I thought I account for that but must have slipped through...now reverted.

**Review by rlmoser99 — APPROVED** [review]
Thanks so much for making these changes so quickly!

**Pull request merged**